numaprom/udf/inference.py

```diff
@@ -77,44 +81,49 @@ def _is_model_stale(
 def inference(_: str, datum: Datum) -> list[bytes]:
     _start_time = time.perf_counter()
 
-    stream_payload = None
-    train_payload = None
     _in_msg = datum.value.decode("utf-8")
 
     # Construct payload object
-    try:
-        stream_payload = StreamPayload(**orjson.loads(_in_msg))
-    except TypeError:
-        train_payload = TrainerPayload(**orjson.loads(_in_msg))
+    payload = PayloadFactory.from_json(_in_msg)
 
     # Check if trainer payload is passed on from previous vtx
-    if train_payload:
+    if isinstance(payload, TrainerPayload):
         _LOGGER.debug("%s - Relaying forward trainer payload")
-        return [orjson.dumps(train_payload)]
+        return [orjson.dumps(payload)]
+
+    # Check if this payload has performed static thresholding
+    if payload.header == Header.STATIC_INFERENCE:
+        _LOGGER.debug("%s - Relaying forward static threshold payload")
+        return [orjson.dumps(payload, option=orjson.OPT_SERIALIZE_NUMPY)]
 
     messages = []
 
     # Load config
-    metric_config = get_metric_config(stream_payload.composite_keys["name"])
+    metric_config = get_metric_config(payload.composite_keys["name"])
     model_config = metric_config["model_config"]
 
     # Check if model exists
-    artifact_data = _get_model(stream_payload, model_config)
+    artifact_data = _get_model(payload, model_config)
     if not artifact_data:
+        msgs = []
         train_payload = TrainerPayload(
-            uuid=stream_payload.uuid, composite_keys=OrderedDict(stream_payload.composite_keys)
+            uuid=payload.uuid, composite_keys=OrderedDict(payload.composite_keys)
         )
-        return [orjson.dumps(train_payload)]
+        msgs.append(orjson.dumps(train_payload))
+
+        # Calculate scores using static threshold
+        msgs.append(calculate_static_thresh(payload, STATIC_LIMIT))
```

Comment:
why are we doing this again here? if STATIC_INFERENCE we are just need to forward to next vertex right? which is done in line 95.

Reply:
We should just do this in threshold vertex, add a new field to `TrainPayload` like `stale_model` may be. If not stale model, that means no model at all, then calculate static threshold and send to postprocess.